sklearn/sklearn-template/template/README.md

```diff
@@ -0,0 +1,105 @@
+- [Google BigQuery](https://cloud.google.com/bigquery/) A fast, highly scalable, cost-effective, and fully managed 
+cloud data warehouse for analytics, with even built-in machine learning.
+
+# Structure of the template
+```
+Template              
+    |__ scripts
+        |__ train.sh            # convenience script for running machine learning training jobs 
+    |__ trainer                 # trainer package
+        |__ metadata.py         # dataset metadata and feature columns definitions
+        |__ model.py            # pre-processing and machine learning model pipeline definition
+        |__ utils.py            # utility functions including e.g. loading data from bigquery and cloud storage
+        |__ task.py             # training job entry point, handling the parameters passed from command line 
+    |__ hptuning_config.yaml    # config file for running hyperparameter tunning 
+    |__ setup.py                # specify necessary dependency of the training job
+    |__ requirements.txt        # specify necessary dependency of the training job
+    |__ Dockerfile
+```
+
+# Steps to make use of the template
+## Step 0. Prerequisite
+Before you follow the instructions below to adapt the tempate to your machine learning job, 
+you need a Google cloud project if you don't have one. You can find detailed instructions 
+[here](https://cloud.google.com/dataproc/docs/guides/setup-project).
+
+- Make sure the following API & Services are enabled.
+    * Cloud Storage
+    * Cloud Machine Learning Engine
+    * BigQuery API
+    * Cloud Build API (for CI/CD integration)
+    * Cloud Source Repositories API (for CI/CD integration)
+
+- Configure project id as environment variable
+  ```bash
+  $ export PROJECT_ID=[your-google-project-id]
+  ```
+  
+## Step 1. Modify metadata.py
+```python
+# Example for iris dataset
+COLUMNS = None  # Schema of the data. Necessary for data stored in GCS
+
+NUMERIC_FEATURES = [
+    'sepal_length',
+    'sepal_width',
+    'petal_length',
+    'petal_width',
+]
+
+CATEGORICAL_FEATURES = [
+
+]
+
+FEATURE_COLUMNS = NUMERIC_FEATURES + CATEGORICAL_FEATURES
+
+LABEL = 'species'
+
+METRIC_FILE_NAME_PREFIX = 'metric'
+MODEL_FILE_NAME_PREFIX = 'model'
+MODEL_FILE_NAME_SUFFIX = '.joblib'
+
+BASE_QUERY = '''
+    Select * From `{table}`
+  '''
+```
+
+metadata.py is where the dataset's metadata is defined. The code snippets above is an example configured 
+for iris dataset (can be found at `bigquery-public-data.ml_datasets.iris`). In most cases, only the following
+items need to be modified, in order to adapt to the target dataset. 
+- **COLUMNS**: the schema of ths data, only required for data stored in GCS
+- **NUMERIC_FEATURES**: columns those will be treated as numerical features
+- **CATEGORICAL_FEATURES**: columns those will be treated as categorical features
+- **LABEL**: column that will be treated as label
+
+## Step 2. Submit ML training job
+```shell
+bash scripts/train.sh [INPUT_PATH] [RUN_ENV] [BUCKET_NAME]
```

Comment:
Regarding BUCKET_NAME, I'm inclined to omit this from the script params as it doesn't really change much during model dev (refer to my comment in PR#9 for train.sh).

Reply:
Modified.